sound/soc/sof/sof-audio.c

```diff
@@ -94,7 +103,39 @@ static void sof_widget_free(struct snd_sof_dev *sdev, struct snd_sof_widget *swi
 	struct sof_ipc_reply reply;
 	int ret;
 
-	/* only free when refcount is 0 */
+	/*
+	 * If it is a HDA/ALH type DAI and a static widget, it's config should also be updated
+	 * every time it is freed and the current use_count is > 1. All other static widgets,
```

Comment:
I'm missing the "why" part here. Why it needs to be updated, and why only for intel?

Reply:
So before we had dynamic pipelines, we were doing this in sof_widget_list_dai_config_setup() but since that has been revmoved the logic is clubbed here.